test/integration/filters/modify_buffer_filter.cc

```diff
@@ -13,7 +13,9 @@ namespace Envoy {
 // the content of the filter buffer.
 class ModifyBufferStreamFilter : public Http::PassThroughFilter {
 public:
-  Http::FilterDataStatus decodeData(Buffer::Instance&, bool end_stream) {
+  Http::FilterDataStatus decodeData(Buffer::Instance& data, bool end_stream) {
+    decoder_callbacks_->addDecodedData(data, true);
```

Comment:
With this addition, I think you want to return no buffer in the response code. Before @alyssawilk yells at me about this, I agree this is too complicated, and I will continue to think about how to make this better generically. :)

Same for the encode case.

Reply:
yeah that makes sense, updated